File: Java/Medium/Accepted/count-number-of-homogenous-substrings_7.java
class Solution {
    public int countHomogenous(String s) {
        
        int count = 0;
        int result = 0;
        char prev = ' ';
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == prev) {
                count++;
            } else {
                count = 1;
                prev = c;
            }
            result += count;
            result %= 1000000007;
        }
        return result;
    }
}

File: Rust/Medium/WrongAnswer/count-number-of-homogenous-substrings_7.rs
impl Solution {
    pub fn count_homogenous(s: String) -> i32 {
        
        let modu = 1_000_000_007;
        let mut count = 0;
        let mut prev = 'a';
        let mut num = 0;
        for c in s.chars() {
            if c != prev {
                prev = c;
                num = 0;
            }
            num += 1;
            count += num;
        }
        (count % modu) as i32
    }
}